README.md

```diff
@@ -1,289 +1,822 @@
+
+The middleware can be configured through the `IApplicationBuilder` or `IEndpointRouteBuilder`
+builder interfaces.  In addition, an `ExecutionResultActionResult` class is added for returning
+`ExecutionResult` instances directly from a controller action.
+
+Authorization is also supported with the included `AuthorizationValidationRule`.  It will
+scan GraphQL documents and validate that the schema and all referenced output graph types, fields of
+output graph types, and query arguments meet the specified policy and/or roles held by the
+authenticated user within the ASP.NET Core authorization framework.  It does not validate
+any policies or roles specified for input graph types, fields of input graph types, or
+directives.  It skips validations for fields or fragments that are marked with the `@skip` or
+`@include` directives.
+
+See [migration notes](migration/migration7.md) for changes from version 6.x.
+
+## Configuration
+
+### Typical configuration with HTTP middleware
 
-> **TL;DR**
-> Install [GraphQL.Server.All](https://www.nuget.org/packages/GraphQL.Server.All) meta package with all the packages you need to get started.
+First add either the `GraphQL.Server.All` nuget package or the `GraphQL.Server.Transports.AspNetCore`
+nuget package to your application.  Referencing the "all" package will include the UI middleware
+packages.  These packages depend on `GraphQL` version 7.0.0 or later.
 
-#### 1. HTTP/WebSockets middleware for GraphQL
+Then update your `Program.cs` or `Startup.cs` to configure GraphQL, registering the schema
+and the serialization engine as a minimum.  Configure WebSockets and GraphQL in the HTTP
+pipeline by calling `UseWebSockets` and `UseGraphQL` at the appropriate point.
+Finally, you may also include some UI middleware for easy testing of your GraphQL endpoint
+by calling `UseGraphQLVoyager` or a similar method at the appropriate point.
 
-For the HTTP/WebSockets middleware:
+Below is a complete sample of a .NET 6 console app that hosts a GraphQL endpoint at
+`http://localhost:5000/graphql`:
 
+#### Project file
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net6.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="GraphQL.Server.All" Version="7.0.0" />
+  </ItemGroup>
+
+</Project>
 ```
-> dotnet add package GraphQL.Server.Transports.AspNetCore
+
+#### Program.cs file
+
+```csharp
+using GraphQL;
+
+var builder = WebApplication.CreateBuilder(args);
+builder.Services.AddGraphQL(b => b
+    .AddAutoSchema<Query>()  // schema
+    .AddSystemTextJson());   // serializer
+
+var app = builder.Build();
+app.UseDeveloperExceptionPage();
+app.UseWebSockets();
+app.UseGraphQL("/graphql");            // url to host GraphQL endpoint
+app.UseGraphQLPlayground(
+    new GraphQL.Server.Ui.Playground.PlaygroundOptions {
```

Comment:
I would prefer to move opening curly brace on new line like it is across codebase.